### PR TITLE
WDPD-215 add saved cards to payment selection

### PR DESCRIPTION
**Pull request by ps-cgrach**
### This PR

Add the saved Credit cards to the payment selection page.

### Note
Address handling is not working correctly. Due to 400 LOC, will be done in the next PR(#174).

### Testing
This branch is not really testable, test final handling in PR(#174)

### Jira Ticket
https://jira.parkside.at/browse/WDPD-215

### CODE REVIEW

File: Model/InvoicePaymentMethod.php [hunk 1]
```diff
@@ -108,7 +108,7 @@ protected function _checkSaveCheckoutFields($aCheckoutFields)
             }
         }
 
-        return $bDataToSave && Registry::getSession()->getUser()->oxuser__oxpassword->value !== '';
+        return $bDataToSave && Registry::getSession()->getUser()->hasAccount();
```
Comment (ps-amaureder) [inline Model/InvoicePaymentMethod.php]: Refactoring, use the already existing function for the check

File: Core/Vault.php [hunk 2]
```diff
@@ -41,7 +42,8 @@ public static function getCards()
     {
         $oUser = Registry::getSession()->getUser();
         $sUserId = $oUser->getId();
-        $sAddressId = $oUser->getSelectedAddressId();
+        //FIXME get shipping address
```
Comment (ps-dfritzsch) [inline Core/Vault.php]: @tomazpu Address handling fixed in another PR to reduce the size of this one (see PR description):
https://github.com/wirecard/oxid-ee/pull/174/files

File: Core/OrderHelper.php [hunk 3]
```diff
@@ -410,4 +414,38 @@ public static function onBeforeOrderCreation($oPayment)
 
         return true;
     }
+
+    /**
+     * Get the last order's shipping address for the current user
```
Comment (ps-holler) [inline Core/OrderHelper.php]: (minor) It's actually the given user, not the current one.

File: Extend/Controller/PaymentController.php [hunk 4]
```diff
@@ -116,4 +119,27 @@ private function _initializeInputFields($oPayment)
...
+     *
+     * @throws DatabaseConnectionException
+     * @throws DatabaseErrorException
+     *
+     * @return mixed
+     * @since 1.3.0
+     */
+    public function validatePayment()
+    {
+        $iDeleteId = Registry::getRequest()->getRequestParameter('wd_delete_id');
```
Comment (ps-holler) [inline Extend/Controller/PaymentController.php]: Maybe we should give this key a more speaking name, e.g. `wd_vault_deletion_id`

File: Model/CreditCardPaymentMethod.php [hunk 5]
```diff
@@ -319,4 +324,170 @@ public function getMetaDataFieldNames()
...
+
+    /**
+     * @param string $sToken
+     * @param bool   $bCheked
+     *
+     * @return string
+     *
+     * @since 1.3.0
+     */
+    private static function _createRadioButton($sToken, $bCheked = false)
```
Comment (ps-holler) [inline Model/CreditCardPaymentMethod.php]:
```suggestion
    private static function _createRadioButton($sToken, $bChecked = false)
```

File: Model/CreditCardPaymentMethod.php [hunk 6]
```diff
@@ -319,4 +324,170 @@ public function getMetaDataFieldNames()
...
+    /**
+     * @param int $iCardId
+     *
+     * @return string
+     *
+     * @since 1.3.0
+     */
+    private static function _createDeleteButton($iCardId)
+    {
+        return '<button class="btn btn-error" type = "submit" name = "wd_delete_id" value = "' . $iCardId . '" />' .
```
Comment (ps-holler) [inline Model/CreditCardPaymentMethod.php]: The spaces before and after the equals sign are unusual and not consistent.

**Pull request merged**